.github/workflows/hardhat-core-ci.yml

```diff
@@ -56,13 +56,13 @@ jobs:
         uses: actions-rs/cargo@v1
         with:
           command: check
-          args: --workspace --all-features --all-targets
+          args: --workspace --all-features --all-targets --profile bench
```

Comment:
I think this means that after this change we're _not_ doing `cargo check --profile dev`, which we _were_ doing before. Is that okay?

![image](https://github.com/NomicFoundation/hardhat/assets/7883777/9323ca5c-2f9c-4a9f-8d5c-96aa4f78bdeb)
https://doc.rust-lang.org/cargo/reference/profiles.html#profile-selection

Reply:
Bench is the most complete profile for this case, if I understand correctly. I could be wrong though:
- test: extends dev but runs #[test]
- bench: extends release but runs #[test] and #[bench]
- dev: unoptimised code
- release: optimised code